### PR TITLE
Fix multiselect markup

### PR DESCRIPTION
**Pull request by alex-ju**
This PR updates the markup generated by the autocomplete for a multiselect instance to:
- Render selected options list only when it has items
- Remove `listbox` role from selected items list
- Ensure `id`s for selected items are unique (as they were the same with the ones in the suggestion list)

[Trello card](https://trello.com/c/NWhCIDOF)

### CODE REVIEW

File: test/functional/wrapper.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ describe('Wrapper', () => {
     const autocompleteList = autocompleteInstance.querySelector('.autocomplete__list')
     expect(autocompleteInput.tagName.toLowerCase()).to.equal('input')
     expect(autocompleteInput.id).to.equal(id)
-    expect(autocompleteList.tagName.toLowerCase()).to.equal('ul')
+    expect(autocompleteList).to.equal(null)
```
Comment (ChrisBAshton) [inline test/functional/wrapper.js:142]: It looks like we're now missing a test for when there _are_ more than zero options 🤔

File: src/autocomplete.js [hunk 2]
```diff
@@ -600,7 +600,7 @@ export default class Autocomplete extends Component {
           >
             {selectedOptions.map((option, index) => {
               return (
-                <li id={`${id}__option--${index}`} className='autocomplete__selected-option'>
+                <li id={`${id}__selected-option--${index}`} className='autocomplete__selected-option'>
```
Comment (ChrisBAshton) [inline src/autocomplete.js:603]: May be worth writing a test so that this doesn't happen again in a future regression?

Along the lines of: `expect(document.querySelector("#foo__option--1").length).toEqual(1)`

**Review by ChrisBAshton — COMMENTED** [review]
Looks good to me 🎉 my only suggestion is adding a couple of tests. (I'm always nervous if a code change doesn't require any corresponding test change!).

File: test/functional/wrapper.js (hunk 1 above)
Comment (alex-ju) [reply test/functional/wrapper.js:142]: We have [one that captures this](https://github.com/kevindew/accessible-autocomplete/blob/master/test/functional/wrapper.js#L183-L184) already. Which could be updated to assert against `__selected-option` to address the second comment.

Comment (ChrisBAshton) [reply test/functional/wrapper.js:142]: Ah, great 👍 is it worth adding the line we've removed here, to that test? (Asserting that the containing element is a `<ul>`?)

Comment (alex-ju) [reply test/functional/wrapper.js:142]: Updated the mentioned test to capture both the element types and the associated classes.

Comment (ChrisBAshton) [reply test/functional/wrapper.js:142]: Looks good 👍

File: src/autocomplete.js (hunk 2 above)
Comment (ChrisBAshton) [reply src/autocomplete.js:603]: I think this still stands ^ could you let me know what you think Alex? You've tested counting elements of a certain class, but this test would be to do with IDs. It's only the duplicate IDs that would cause a11y issues.
